subprojects/docs/src/docs/userguide/tasks.xml

```diff
@@ -290,52 +290,673 @@
+                            annotation, just as we would with a <literal>String</literal> property.
+                        </para>
+                    </listitem>
+                    <listitem><para><literal>sourceFiles</literal></para>
+                        <para>The source templates that the task will be processing. Single files and
+                            collections of files need their own special annotations. In this case, we’re
+                            dealing with a collection of input files and so we use the
+                            <literal>@InputFiles</literal> annotation. You’ll see more file-oriented
+                            annotations in a table later.
+                        </para>
+                    </listitem>
+                    <listitem><para><literal>templateData</literal></para>
+                        <para>For this example, we’re using a custom class to represent the model data.
+                            However, it does not implement <literal>Serializable</literal>, so we can’t
+                            use the <literal>@Input</literal> annotation. That’s not a problem as the
+                            properties within <literal>TemplateData</literal> - a string and a hash map
+                            with serializable type parameters - are serializable and can be annotated with
+                            <literal>@Input</literal>. We use <literal>@Nested</literal> on
+                            <literal>templateData</literal> to let Gradle know that this is a value with
+                            nested input properties.
+                       </para>
+                    </listitem>
+                    <listitem><para><literal>outputDir</literal></para>
+                        <para>The directory where the generated files go. As with input files, there are
+                            several annotations for output files and directories. A property representing
+                            a single directory requires <literal>@OutputDirectory</literal>. You’ll learn
+                            about the others soon.
+                        </para>
+                    </listitem>
+                </itemizedlist>
+
+                <para>These annotated properties mean that Gradle will skip the task if none of the source
+                    files, template engine, model data or generated files have changed since the previous
+                    time Gradle executed the task. This will often save a significant amount of time. You
+                    can learn how Gradle detects <xref linkend="sec:how_does_inc_build_work">changes later</xref>.
+                </para>
+
+                <para>This example is particularly interesting because it works with collections of source
+                    files. What happens if only one source file changes? Does the task process all the
```

Comment:
It's mentioned (and linked) in the advanced incremental build section. I'll think about making it more prominent by moving it to an earlier bit. I'm not sure at this point whether it would act as a distraction or not.